### PR TITLE
In element connect

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?

Lines only connected to 

Issue Number: https://github.com/gaphor/gaphor/issues/390

### What is the new behavior?

Lines glue to an element border, even when the line-end is dragged inside the element.

For this purpose `ConnectionSink` is extended. `ConnectionSink` is now in charge of providing glue points and constraints.

### Does this PR introduce a breaking change?
- [X] Yes
- [ ] No

### Other information

I wonder to what extend we still need ports, since we should be fine with handles and connection sinks.

### CODE REVIEW

File: gaphas/aspect/connector.py [hunk 1]
```diff
@@ -88,16 +107,74 @@ def disconnect(self) -> None:
...
 
+@Connector.register(Line)
+class LineConnector(ItemConnector):
+    def secondary_handle(self) -> Optional[Handle]:
+        item = self.item
+        handle = self.handle
+        handles = item.handles()
+        if len(handles) < 2:
+            return None
+        if handle is item.handles()[0]:
```
Comment (danyeaw) [inline gaphas/aspect/connector.py:118]: Could this be simplified to `if handle is handles[0]`:?

**Review by danyeaw — APPROVED** [review]
This is looks really great, the highlighting of the whole element while over and the snap to the border is on :fire:!

When trying to run the demo:
```
/home/dan/Projects/gaphas/gaphas/view/gtkview.py:7: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk, GLib, GObject, Gtk
/home/dan/Projects/gaphas/gaphas/view/gtkview.py:7: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk, GLib, GObject, Gtk
```

File: gaphas/aspect/connector.py (hunk 1 above)
Comment (amolenaar) [reply gaphas/aspect/connector.py:118]: Yes of course :)

**Comment by amolenaar** [comment]
> This is looks really great, the highlighting of the whole element while over and the snap to the border is on fire!
> 
> When trying to run the demo:
> 
> ```
> /home/dan/Projects/gaphas/gaphas/view/gtkview.py:7: PyGIWarning: Gdk was imported without specifying a version first. Use gi.require_version('Gdk', '3.0') before import to ensure that the right version gets loaded.
>   from gi.repository import Gdk, GLib, GObject, Gtk
> /home/dan/Projects/gaphas/gaphas/view/gtkview.py:7: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
>   from gi.repository import Gdk, GLib, GObject, Gtk
> ```

I removed a `gi.require_version` from the Gaphas code base. I can update the demo to fix this. At least it will not crash, once people have GTK 4 installed on there systems.

File: gaphas/aspect/connector.py (hunk 1 above)
Comment (amolenaar) [reply gaphas/aspect/connector.py:118]: Yes, it should :)

**Pull request merged**